ai-player.js

```diff
@@ -235,10 +235,46 @@ function createAiPlayer(game, options) {
         if (state.state.name == stateNames.FINAL_ACTION_RESPONSE) {
             return false;
         }
+
+        // Challenge if somebody claims to have role that was revealed 3 times or we have the rest of them
+        var usedRoles = countRevealedRoles(state.state.action);
```

Comment:
This seems wrong: `state.state.action` is an action, e.g., `assassinate`, but `countRevealedRoles` accepts a role name, e.g., `assassin`. Also, on the line below, you are comparing a role with an action.